pkg/controller/history/controller_history.go

```diff
@@ -253,6 +253,11 @@ func (rh *realHistory) CreateControllerRevision(parent metav1.Object, revision *
 		clone.Name = ControllerRevisionName(parent.GetName(), hash)
 		created, err := rh.client.AppsV1().ControllerRevisions(parent.GetNamespace()).Create(clone)
 		if errors.IsAlreadyExists(err) {
+			existed, err := rh.client.AppsV1().ControllerRevisions(parent.GetNamespace()).Get(parent.GetName(), metav1.GetOptions{})
```

Comment:
This should use the name of the controller revision rather than the name of the parent stateful set as the first parameter in the Get call. So `Get(clone.Name, metav1.GetOptions{})` instead of `Get(parent.GetName(), metav1.GetOptions{})`